graphql/cxs-impl/src/main/java/org/apache/unomi/graphql/condition/ProfileConditionFactory.java

```diff
@@ -0,0 +1,246 @@
+            }
+        }
+
+        rootCondition.setParameter("subConditions", rootSubConditions);
+        return rootCondition;
+    }
+
+    public Condition createProfileFilterInputCondition(CDPProfileFilterInput filterInput, Date after, Date before, DefinitionsService definitionsService) {
+        final Condition rootCondition = createBoolCondition("and", definitionsService);
+        final List<Condition> rootSubConditions = new ArrayList<>();
+
+        if (after != null) {
+            rootSubConditions.add(createDatePropertyCondition("timeStamp", "greaterThan", after, definitionsService));
+        }
+
+        if (before != null) {
+            rootSubConditions.add(createDatePropertyCondition("timeStamp", "lessThanOrEqual", before, definitionsService));
+        }
+
+        if (filterInput != null) {
+            if (filterInput.getProfileIDs_contains() != null && filterInput.getProfileIDs_contains().size() > 0) {
+                rootSubConditions.add(createPropertiesCondition("_id", "hasSomeOf", filterInput.getProfileIDs_contains(), definitionsService));
+            }
+
+            if (filterInput.getConsents_contains() != null && filterInput.getConsents_contains().size() > 0) {
+                rootSubConditions.add(createPropertiesCondition("consents", "hasSomeOf", filterInput.getConsents_contains(), definitionsService));
+            }
+
+            if (filterInput.getSegments_contains() != null && filterInput.getSegments_contains().size() > 0) {
+                rootSubConditions.add(createPropertiesCondition("segments", "hasSomeOf", filterInput.getSegments_contains(), definitionsService));
+            }
+
+            if (filterInput.getLists_contains() != null && filterInput.getLists_contains().size() > 0) {
+                rootSubConditions.add(createPropertiesCondition("lists", "hasSomeOf", filterInput.getLists_contains(), definitionsService));
+            }
+
+            if (filterInput.getProperties() != null) {
+                rootSubConditions.add(createProfilePropertiesFilterInputCondition(filterInput.getProperties(), definitionsService));
+            }
+
+            if (filterInput.getEvents() != null) {
+                rootSubConditions.add(createProfileEventsFilterInputCondition(filterInput.getEvents(), definitionsService));
+            }
+
+            if (filterInput.getInterests() != null) {
+                rootSubConditions.add(createInterestFilterInputCondition(filterInput.getInterests(), definitionsService));
+            }
+        }
+
+        rootCondition.setParameter("subConditions", rootSubConditions);
+        return rootCondition;
+    }
+
+    private Condition createInterestFilterInputCondition(CDPInterestFilterInput filterInput, DefinitionsService definitionsService) {
+        final Condition rootCondition = createBoolCondition("and", definitionsService);
+        final List<Condition> rootSubConditions = new ArrayList<>();
+
+        if (filterInput.getTopic_equals() != null) {
```

Comment:
That will not works
Interest is a map, where topic is a key, score is a value, for example:
```
"properties": {
    "interests": {
      "interestName1": 10,
      "interestName2": 5,
      "interestName3": 7
    }
  }
```